### ISSUE TITLE
dspace-oai can't inject dependencies from dspace-api

### ISSUE THREAD

**Issue by mwoodiupui** [DSpace/DSpace#8112]
Attempts to @Autowired or @Inject dependencies to dspace-oai from dspace-api fail at runtime.  This seems to be because org.dspace.app.xoai.XOAI builds its ApplicationContext solely from its own configuration classes and ignores the XML configuration.

We should be combining the configurations, as shown in https://docs.spring.io/spring-javaconfig/docs/1.0.0.M4/reference/html/ch06.html 

**To Reproduce**
1. Do this
2. Then this...

**Expected behavior**
DI should work across artifacts.

**Comment by tdonohue** [comment]
Agreed, this would be very beneficial and avoid mistakes like in #8110 (this isn't the only time we forget dspace-oai doesn't support DI properly, but it's the latest time). 

Since you've claimed this ticket @mwoodiupui , I'll pull this onto the 7.3 board in the hopes we can get this fixed by that release (if not though, no worries)
